Source/Scene/Scene.js

```diff
@@ -2696,13 +2750,166 @@ define([
         return object;
     };
 
+    var fragDepthRegex = /\bgl_FragDepthEXT\b/;
+    var discardRegex = /\bdiscard\b/;
+
+    function getDepthOnlyShaderProgram(scene, context, shaderProgram) {
+        var id = shaderProgram.id;
+        var cache = scene._depthOnlyShaderProgramCache;
+        var shader = cache[id];
+        if (!defined(shader)) {
+            var attributeLocations = shaderProgram._attributeLocations;
+            var fs = shaderProgram.fragmentShaderSource;
+
+            var writesFragDepth = false;
```

Comment:
Perhaps combine this into one: `writesDepthOrDiscards` and simplify the logic below?